include/neml2/models/crystallography/MillerIndex.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include "neml2/tensors/FixedDimTensor.h"
+
+#include <string>
+
+namespace neml2
+{
+class Vec;
+namespace crystallography
+{
+/// Represention of a crystal direction or plane a Miller Index
+// I delibrately made this not inherit from VecBase but we could
+// reconsider that later on
+class MillerIndex : public FixedDimTensor<MillerIndex, 3>
```

Comment:
I'd like to understand why we don't want MillerIndex to derive from VecBase. Next question, does MillerIndex have to be a class?

Reply:
I don't want `Vec` classes to mix with `MillerIndex`

It needs to be a class, not now, but we will have a subclass for `MillerBravais` indices eventually.